src/main/scala/com/imaginea/activegrid/core/models/ChangeDirectoryCommand.scala

```diff
@@ -0,0 +1,25 @@
+package com.imaginea.activegrid.core.models
+
+import com.beust.jcommander.Parameter
+import com.imaginea.activegrid.core.utils.CmdExecUtils
+
+import scala.collection.JavaConversions._ // scalastyle:ignore underscore.import
+
+/**
+  * Created by shareefn on 16/12/16.
+  */
+class ChangeDirectoryCommand extends Command {
+
+  @Parameter(description = "the new context to change to")
+  var newContext: java.util.List[String] = new java.util.ArrayList()
+
+  override def execute(commandExecutionContext: CommandExecutionContext, lines: List[Line]): CommandResult = {
+    val context = newContext.size match {
+      case 1 =>
```

Comment:
we are getting newContext after JCommander parsing. If newContext(a list) contains one element  (cd command requires only one argument), we pass it to CmdExecUtils and get new context . otherwise we return current context.